src/main/java/io/github/bobocodebreskul/server/WebPathScanner.java

```diff
@@ -0,0 +1,167 @@
+    RequestMapping requestMapping = annotation.annotationType().getAnnotation(RequestMapping.class);
+    return getHttpMethodName(requestMapping);
+  }
+
+  private static void addPath(Map<String, Map<String, ControllerMethod>> allPath, String path,
+      String httpMethodName, ControllerMethod controllerMethod) {
+
+    if (allPath.containsKey(path)) {
+      Map<String, ControllerMethod> httpMethodControllerMethodMap = allPath.get(path);
+      if (httpMethodControllerMethodMap.containsKey(httpMethodName)) {
+        throw new DuplicatePathException(
+            "Duplicate path %s for http method %s detected".formatted(path, httpMethodName));
+      } else {
+        httpMethodControllerMethodMap.put(httpMethodName, controllerMethod);
+      }
+    } else {
+      allPath.put(path, new HashMap<>(Map.of(httpMethodName, controllerMethod)));
+    }
+  }
+
+  /**
+   * This method is responsible for scanning and creating web paths. It uses bring container to
+   * retrieve all Beans marked as {@link RestController} then retrieve path and http method from
+   * public methods marked as {@link Post}, {@link Get}, {@link Delete}, {@link Put}, {@link Head},
+   * {@link RequestMapping}
+   *
+   * @return Map of (path, Map of (HTTP method, ControllerMethod))
+   * @throws IllegalAccessException    if this {@code Method} object is enforcing Java language
+   *                                   access control and the underlying method is inaccessible.
+   * @throws IllegalArgumentException  if the method is an instance method and the specified object
+   *                                   argument is not an instance of the class or interface
+   *                                   declaring the underlying method (or of a subclass or
+   *                                   implementor thereof); if the number of actual and formal
+   *                                   parameters differ; if an unwrapping conversion for primitive
+   *                                   arguments fails; or if, after possible unwrapping, a
+   *                                   parameter value cannot be converted to the corresponding
+   *                                   formal parameter type by a method invocation conversion.
+   * @throws InvocationTargetException if the underlying method throws an exception.
+   */
+  public Map<String, Map<String, ControllerMethod>> getAllPaths()
+      throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+    Map<String, Map<String, ControllerMethod>> pathMap = new HashMap<>();
+
+    for (Object controllerBean : getControllerBeans()) {
+      var controllerClass = controllerBean.getClass();
+      String prefixPath = getPrefixPath(controllerClass);
+
+      for (Method method : controllerClass.getMethods()) {
+        Annotation httpMethodAnnotation = getHttpAnnotation(method);
+
+        validateIfMethodHasOnlyRequestMappingOrHttpMethodMapping(method, httpMethodAnnotation);
```

Comment:
I think it's more convenient to do the validation in `getHttpAnnotation` method than here